### PR TITLE
Move render common and help functions to common package

### PR DESCRIPTION
**Pull request by Brian-McM**
## Description

This is step one in refactoring the render package into a more manageable structure. the render/common package contains utility packages responsible for rendering k8s objects. For instance, the render/common/meta package contains utilities for creating annotations and getting namespace names, and the render/common/elasticsearch package contains decorators to add elasticsearch specific items to all the components that interact with elasticsearch.

The next step is to move every component in the render package into it's own subpacakge, i.e. render/node, render/logstorage, and break cross component access.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by caseydavenport — COMMENTED** [review]
Nothing here looks too controversial. I would be interested in seeing the final package structure spelled out a bit more so that I understand what this is building towards a bit better.

The only other thing I noticed was this:

> 	rutil "github.com/tigera/operator/pkg/render/common"

"common" is a bit of a flag that this package will be a dumping ground for stuff that we don't know where to put anywhere else. Sometimes that's necessary, but maybe we could do better given the purpose of this is to reorganize the package structure to have better-defined packages?

**Comment by Brian-McM** [comment]
> "common" is a bit of a flag that this package will be a dumping ground for stuff that we don't know where to put anywhere else.

**Comment by Brian-McM** [comment]
@caseydavenport maybe we should hold this for a minute, I'll try to come up with the complete structure instead of a partial one before moving forward.

File: pkg/render/common/data/secrets.go
Comment (tmjd) [inline pkg/render/common/data/secrets.go:1]: I'm not a fan of `common/data`, data to me suggests that it just contains some common "data".
I don't feel like the Secret/TLS creation and resource copying fits, as "data". If we want to keep everything under data in one package it feels like a "render/utils" would be most appropriate for what is in "data" currently. I think if we can't come up with a better package name then I think "utils" is more descriptive than "data".
Saying that I think perhaps we should have a `render/common/secret` package and then any secret manipulation (including CreateTLSSecret) can go in there.

File: pkg/render/common/data/convenience.go
Comment (tmjd) [inline pkg/render/common/data/convenience.go:31]: If we have a `render/common/secret` package I think this should go in that package.

File: pkg/render/common/data/security.go
Comment (tmjd) [inline pkg/render/common/data/security.go:1]: I think we should switch these into their own package of `render/common/security`. I'd even consider splitting the functions into their own packages of `render/common/podsecuritypolicy` and `render/common/securitycontext`.

File: pkg/render/compliance.go
Comment (tmjd) [inline pkg/render/compliance.go:34]: WDYT about `res` for this? `relasticsearch` feels a bit long and it reads like a typo to me but that could just be me, personally I'd rather it be an acronym than something that reads "odd".
I think this is just preference so no change necessary.

**Review by tmjd — COMMENTED** [review]
I didn't do a thorough review but here are a few thoughts.

File: pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go
Comment (caseydavenport) [inline pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go:22]: Do we need to alias this? We don't have any other packages named meta imported here?

File: pkg/controller/utils/elasticsearch.go
Comment (caseydavenport) [inline pkg/controller/utils/elasticsearch.go:105]: I think this function should stay the same name, or be moved into an ES specific package, otherwise it's too vague for a "utils" package. 

Also the comment seems a little wonky now?

File: pkg/controller/logcollector/logcollector_controller.go
Comment (caseydavenport) [inline pkg/controller/logcollector/logcollector_controller.go:43]: same here - why do we need to alias everything with an "r"?

File: pkg/render/compliance.go
Comment (caseydavenport) [reply pkg/render/compliance.go:34]: Why not just "elasticsearch", or if we really feel we need to alias it do "es"? I'm not sure we need the alias, though.

I don't like "res" because that's a common abbreviation for "resource".

**Review by caseydavenport — COMMENTED** [review]
nifty

File: pkg/controller/utils/elasticsearch.go
Comment (Brian-McM) [reply pkg/controller/utils/elasticsearch.go:105]: Ya I did some global search and replace, I missed a few of these.

File: pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go
Comment (Brian-McM) [reply pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go:22]: We don't _need_ to, I just thought having a consistent package name would be nice to have.

File: pkg/controller/logcollector/logcollector_controller.go
Comment (Brian-McM) [reply pkg/controller/logcollector/logcollector_controller.go:43]: The "r" is "render", I thought it would be nicer to have that the name consistent and differentiate between the elasticsearch package which actually does stuff with elasticsarch. It's kind of like how we alias the k8s core package from /v1 to corev1 even if there's nothing conflicting.

File: pkg/controller/compliance/compliance_controller_test.go
Comment (Brian-McM) [reply pkg/controller/compliance/compliance_controller_test.go:184]: Because I don't think we should be generating tls certificates in the render package, I'd like to separate these concerns. We can achieve the same thing by having a convenience function in the controller that creates the certificate and passes it to the render function, but I just chose to do this instead since it since it was just an extra line. I can make a convenience function if you'd like in the controllers package.

File: pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go
Comment (caseydavenport) [reply pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go:22]: gotcha - maybe we should keep our own packages with the name that we give them, and alias any externally conflicting packages? 

I'm guessing this is conflicting with the k8s meta package, which is normally imported as `metav1` IIRC?

File: pkg/render/common/data/secrets.go
Comment (Brian-McM) [reply pkg/render/common/data/secrets.go:1]: I agree, I'm not the hugest fan of data either, I just don't like "utils" either. Maybe we could just have a secrets package, but I'm unsure about adding a package for everything else that's in here (some constants not secret related, config map stuff, pod security).

What do you think, should I just rename it to utils or create a bunch of separate packages that may not have much in them? I'm a little torn...

Comment (Brian-McM) [reply pkg/render/common/data/secrets.go:1]: I just saw your comment below being in favour of multiple packages. I'll change that.

File: pkg/render/common/data/security.go
Comment (Brian-McM) [reply pkg/render/common/data/security.go:1]: Will do

File: pkg/render/compliance.go
Comment (Brian-McM) [reply pkg/render/compliance.go:34]: It doesn't feel too long to me, but I think I'm more OK with longer names than most developers. @caseydavenport I'd like to keep the prefix so that it is clear that this is for rendering Elasticsearch components and not working with Elasticsearch, even if there isn't a package collision. It just keeps things consistent among all packages.

File: pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go
Comment (Brian-McM) [reply pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go:22]: Ya that's a good idea, but for Elasticsearch we have multiple packages of our own. I wanted to keep the original name for the package that actually does stuff with elasticsearch, and alias the rendering package.

**Review by caseydavenport — COMMENTED** [review]
This LGTM - @tmjd should have the final say

File: pkg/controller/compliance/compliance_controller_test.go
Comment (tmjd) [reply pkg/controller/compliance/compliance_controller_test.go:184]: I'm talking about why do MakeCA when CreateTLSSecret will do the call for us if we pass nil CreateTLSSecret.
I don't understand how directly doing MakeCA here is a step in the direction you're suggesting of moving cert creation out of render.

Comment (Brian-McM) [reply pkg/controller/compliance/compliance_controller_test.go]: I've reverted this change as per our conversation on zoom and we'll circle back on how to handle this later.

File: pkg/render/common/secret/secrets.go
Comment (tmjd) [inline pkg/render/common/secret/secrets.go:18]: nit: Doesn't match function name.

**Review by tmjd — COMMENTED** [review]
One nit but LGTM. I don't need to see it again.

File: pkg/render/common/secret/secrets.go
Comment (Brian-McM) [reply pkg/render/common/secret/secrets.go]: updated

**Review by tmjd — APPROVED** [review]

**Pull request merged**
